### PR TITLE
Refactor Dropdown & minor fix TS errors

### LINKED ISSUE #257: Fix linter errors from existing components

### PR DESCRIPTION
**Pull request by anbraten**
closes #257

### CODE REVIEW

File: packages/app/src/components/uiBlocks/Dropdown.vue [hunk 1]
```diff
@@ -103,76 +96,53 @@ export default defineComponent({
       required: false,
       default: '',
     },
+    items: {
+      type: Array as PropType<ListItem[]>,
+      default: (): ListItem[] => [],
```
Comment (julianhahn) [inline packages/app/src/components/uiBlocks/Dropdown.vue:101]: does this syntax creates an new instace?

File: packages/app/src/components/uiBlocks/Dropdown.vue [hunk 2]
```diff
@@ -83,15 +77,14 @@ export default defineComponent({
       required: false,
       default: 'Please select an item',
     },
-    addItemOption: {
+    enableAddItem: {
```
Comment (JanEberlein) [inline packages/app/src/components/uiBlocks/Dropdown.vue:80]: good name 👍

**Review by julianhahn — APPROVED** [review]
okay.

File: packages/app/src/components/uiBlocks/Dropdown.vue (hunk 1 above)
Comment (anbraten) [reply packages/app/src/components/uiBlocks/Dropdown.vue:101]: yeah, a new empty array of type ListItem.

**Review by JanEberlein — APPROVED** [review]
very nice 😄

**Pull request merged**